syntax = "proto3";

package library;

service BookService {
  rpc AddBook(BookRequest) returns (BookResponse);
  rpc ReserveBook(ReserveRequest) returns (BookResponse);
  rpc GetBooks(Empty) returns (BookList);
  rpc ReturnBook (ReturnRequest) returns (BookResponse);
}

message BookRequest {
  string title = 1;
  string author = 2;
}

message ReserveRequest {
  int32 book_id = 1;
  string username = 2;
}

message BookResponse {
  bool success = 1;
  string message = 2;
}

message Empty {}

message Book {
  int32 id = 1;
  string title = 2;
  string author = 3;
  bool available = 4;
  string reserved_by = 5;
}


message BookList {
  repeated Book books = 1;
}
message ReturnRequest {
  int32 book_id = 1;
  string username = 2;
}
